packages/query-core/src/types.ts

```diff
@@ -118,6 +124,7 @@ export interface QueryOptions<
   networkMode?: NetworkMode
   gcTime?: number
   queryFn?: QueryFunction<TQueryFnData, TQueryKey, TPageParam>
+  persister?: QueryPersister<TQueryFnData, TQueryKey>
```

Comment:
@TkDodo Regarding setting persister as default globally, we could use this new option, and instead of wrapping `queryFn` just provide it on the side when necessary, or globally.